sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -328,6 +329,8 @@ async def serialize_property(
     input_transformer: Optional[Callable[[str], str]] = None,
     typ: Optional[type] = None,
     keep_output_values: Optional[bool] = None,
+    resource_obj=None,
+    property_key=None,
```

Comment:
Should these really ever be None? 
I'd think it should either be
A) Yes, some places call serialize_property and we're not in a resource and/or don't have key. In which case the raise ValueError on line 587 need to handle all those cases [obj&key, just obj, just key, neither].
B) No, everywhere should be able to pass obj and/or key and so they shouldn't default to None and shouldn't be typed as Optionals.

Reply:
Might be just one or the other for the above. Like maybe everywhere can pass "key" and it should just be of type string and not have a default, but resource really does need to be optional and default to None. Might be both need to be optional, might be neither.

Reply:
Yes! I tried passing both as positional arguments, just as you said we are calling the function in some places without any Resource and also we need to pass "key" and "object" in all the function call in that case. So I tried `=None` for defaults.

Reply:
In conclusion, do we want resource object default as `None` and "key" as Optional?

Reply:
Got it!